src/libstd/fs.rs

```diff
@@ -663,15 +623,17 @@ impl Permissions {
     }
 }
 
-#[unstable(feature = "file_type", reason = "recently added API")]
 impl FileType {
```

Comment:
I probably missed the discussion but why is this not an (extensible) enum? This seems like a really good candidate.